### ISSUE TITLE
Google Sheet Sync throws error - Glific Exception **(Protocaol.Undefined

### ISSUE THREAD

**Issue by mohitsharma-22** [glific/glific#3163]
**Describe the bug**
TAP reported this issue. One of their Google sheets is not getting synced and showing error message.
<img width="1000" alt="Screenshot 2023-10-23 at 3 49 31 PM" src="https://github.com/glific/glific/assets/16541548/e430685b-663c-436b-857d-11f27b3c4ae1">

Checked the notification section. Didn't get any warnings.

Tried to save the Google sheet interaction to figure out the cause and received an error.
<img width="682" alt="Screenshot 2023-10-23 at 3 48 22 PM" src="https://github.com/glific/glific/assets/16541548/fd21778e-ac13-4294-8fbb-b90369a573ba">

<img width="1470" alt="Screenshot_2023-10-23_at_1 41 44_PM" src="https://github.com/glific/glific/assets/16541548/2d3afd86-b638-430b-9588-6c6ec7da1552">

Replicated on CC with the same Google sheet 
https://docs.google.com/spreadsheets/d/1K4kBW2x-YiyqU_7in0oXV_o4WWSP2n3Q1tMmHKhAUro/edit#gid=0

<img width="689" alt="Screenshot 2023-10-23 at 3 52 30 PM" src="https://github.com/glific/glific/assets/16541548/e783393a-f93d-463d-8968-70ea5d54d7d8">

**To Reproduce**
1. Go to Glific
2. Open Google sheet
3. Add new sheet 
4. Click on SAVE button

**Comment by dlobo** [comment]
@mohitsharma-22 do u have any more details on this. any idea how the characteristics of this sheet?

**Comment by mohitsharma-22** [comment]
@dlobo No, I didn't any suspicious thing in the google sheet, shared by TAP.(https://docs.google.com/spreadsheets/d/1K4kBW2x-YiyqU_7in0oXV_o4WWSP2n3Q1tMmHKhAUro/edit#gid=1270677768)
